### PR TITLE
Add the ability to approve a deposit in the pending approval state

### LINKED ISSUE #352: Approve an item
As a collection reviewer I should be able to view and approve for deposit an item awaiting review in a collection that I am a reviewer of

See https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/420955157

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Fixes #352 

## How was this change tested?

Screen Shot 2020-11-11 at 3 49 10 PM

### CODE REVIEW

File: spec/components/works/approval_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      expect(rendered.to_html).to be_blank
+    end
+  end
+
+  context 'when needing approval' do
+    before do
+      allow(controller).to receive(:allowed_to?).and_return(true)
+    end
+
+    it 'renders nothing' do
```
Comment (mjgiarlo) [inline spec/components/works/approval_component_spec.rb:25]: Copypasta description?

File: app/views/works/show.html.erb [hunk 2]
```diff
@@ -1,267 +1,232 @@
...
-          <th scope="col"></th>
-        </tr>
-        </thead>
-        <tbody>
-        <% if @work.purl %>
+  <section class="details">
+    <header class="title">
+      <span class="header-text"><%= @work.title %></span>
+      <%= link_to edit_work_path(@work), aria: { label: "Edit #{@work.title}"} do %><span class="fas fa-pencil-alt edit"></span><% end %>
+      <% if @work.state != 'deposited' %><span class="font-italic" style="margin-left: 30px;">Draft - Not deposited</span><% end %>
```
Comment (mjgiarlo) [inline app/views/works/show.html.erb:8]: I think you can do `unless @work.deposited?` here too if you're cool with it. Yay for magic state machine methods.

**Review by mjgiarlo — COMMENTED** [review]
Couple quick comments

File: app/views/works/show.html.erb (hunk 2 above)
Comment (jcoyne) [reply app/views/works/show.html.erb:8]: Sure, this isn't really code I changed though (other than indent level).

Comment (mjgiarlo) [reply app/views/works/show.html.erb]: That'll teach you to indent.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**